rust/src/operations/update.rs

```diff
@@ -0,0 +1,921 @@
+        assert_eq!(table.version(), 0);
+        assert_eq!(table.get_file_uris().count(), 1);
+
+        let (table, metrics) = DeltaOps(table)
+            .update()
+            .with_update("value", col("value") + lit(1))
+            .await
+            .unwrap();
+
+        assert_eq!(table.version(), 1);
+        assert_eq!(table.get_file_uris().count(), 1);
+        assert_eq!(metrics.num_added_files, 1);
+        assert_eq!(metrics.num_removed_files, 1);
+        assert_eq!(metrics.num_updated_rows, 5);
+        assert_eq!(metrics.num_copied_rows, 0);
+
+        let expected = [
+            "+-------+",
+            "| value |",
+            "+-------+",
+            "|       |",
+            "|       |",
+            "| 1     |",
+            "| 3     |",
+            "| 5     |",
+            "+-------+",
+        ];
+
+        let actual = get_data(&table).await;
+        assert_batches_sorted_eq!(&expected, &actual);
+
+        // Validate order operators do not include nulls
+        let table = prepare_values_table().await;
+        let (table, metrics) = DeltaOps(table)
+            .update()
+            .with_predicate(col("value").gt(lit(2)).or(col("value").lt(lit(2))))
+            .with_update("value", lit(10))
+            .await
+            .unwrap();
+        assert_eq!(table.version(), 1);
+        assert_eq!(table.get_file_uris().count(), 1);
+        assert_eq!(metrics.num_added_files, 1);
+        assert_eq!(metrics.num_removed_files, 1);
+        assert_eq!(metrics.num_updated_rows, 2);
+        assert_eq!(metrics.num_copied_rows, 3);
+
+        let expected = [
+            "+-------+",
+            "| value |",
+            "+-------+",
+            "|       |",
+            "|       |",
+            "| 2     |",
+            "| 10    |",
+            "| 10    |",
+            "+-------+",
+        ];
+        let actual = get_data(&table).await;
+        assert_batches_sorted_eq!(&expected, &actual);
+
+        let table = prepare_values_table().await;
+        let (table, metrics) = DeltaOps(table)
+            .update()
+            .with_predicate(col("value").is_null())
+            .with_update("value", lit(10))
+            .await
+            .unwrap();
+        assert_eq!(table.version(), 1);
+        assert_eq!(table.get_file_uris().count(), 1);
+        assert_eq!(metrics.num_added_files, 1);
+        assert_eq!(metrics.num_removed_files, 1);
+        assert_eq!(metrics.num_updated_rows, 2);
+        assert_eq!(metrics.num_copied_rows, 3);
+
+        let expected = [
+            "+-------+",
+            "| value |",
+            "+-------+",
+            "| 10    |",
+            "| 10    |",
+            "| 0     |",
+            "| 2     |",
+            "| 4     |",
+            "+-------+",
+        ];
+        let actual = get_data(&table).await;
+        assert_batches_sorted_eq!(&expected, &actual);
+    }
+
+    #[tokio::test]
+    async fn test_str_expressions() {}
```

Comment:
This a TODO?

Reply:
Yeah my bad. I wanted to add a test to demonstrate that a str can be used for the predicate and update expression but it seemed like a test with little value.

I've updated the null tests to that.
FYI using strings for expressions will require additional work in future PRs.
If you have a predicate like `value < 2 or value > 2` DataFusion will return an error about being unable to compare `int32` to `int64`

Reply:
> If you have a predicate like value < 2 or value > 2 DataFusion will return an error about being unable to compare int32 to int64

I stumbled across this one as well, not sure if this is maybe even a datafusion bug, since the expression parser seems to ignore the information in the schema passed to it.